src/cli/user/login.rs

```diff
@@ -14,6 +14,8 @@ pub struct Options {
     oidc_domain: Url,
     #[clap(long, default_value = "4NuaJxkQv8EZBeJKE56R57gKJbxrTLG2")]
     oidc_client_id: String,
+    #[clap(long)]
+    oidc_client_secret: Option<String>,
```

Comment:
Do we really want to expose this on the command-line?
Especially if this is to be used by CI, using an env variable for this is probably the lesser of all evils

Reply:
I personally think having the behaviour of a command radically change to where it does something pretty different (exchange client credentials for access token, instead of print a device key, waiting for that, and then exchanging that for an access token) because of the existence (or non-existence) of a "magical" environment variable is kinda irky.
At least as a CLI flag, it's very explicit.

But if you really prefer an env var, sure.

Reply:
We should not encourage users to pass secrets on the command line.
This secret belongs in a config file (refs #2021 I suppose).
In CI, writing a config file is cumbersome and an env var is far more common and simpler to use.

Regarding the different behavior - I agree, so it seems like we need another CLI flag, which would enable `m2m` login. It should probably be an enum and default to something like `--login device`, `--login client` being the other option for now.